java-frontend/src/main/java/org/sonar/java/resolve/Resolve.java

```diff
@@ -512,6 +519,17 @@ private boolean isAcceptableType(JavaType arg, JavaType formal, boolean autoboxi
     return types.isSubtype(arg.erasure(), formal.erasure()) || (autoboxing && isAcceptableByAutoboxing(arg, formal.erasure()));
   }
 
+  private static boolean subtypeOfTypeVar(JavaType arg, JavaType.TypeVariableJavaType formal) {
+    for (JavaType bound : formal.bounds()) {
+      if(bound.isTagged(JavaType.TYPEVAR) && !subtypeOfTypeVar(arg, ((JavaType.TypeVariableJavaType) bound))) {
+        return false;
+      } else if (!arg.isSubtypeOf(bound)) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) This branch's code block is the same as the block for the branch on line 524. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1871)